generator/src/lib.rs

```diff
@@ -3222,6 +3223,11 @@ pub fn write_source_code<P: AsRef<Path>>(vk_headers_dir: &Path, src_dir: P) {
             s.elements
                 .iter()
                 .filter_map(get_variant!(vkxml::StructElement::Member))
+                .filter(|x| {
+                    x.name.as_deref() == Some("pNext")
+                        || x.size.is_some()
+                        || !is_opaque_type(&x.basetype)
```

Comment:
The reason I picked this predicate is because no builder pattern currently passes such "opaque pointers" (mostly window and compositor handles) as borrows with a lifetime, since they've generally "opaque" low-level raw pointers to applications as well.

We just don't see it in the diff because any struct containing them appears to already contain other pointers to Rust-defined structures that are affecting the lifetime of the struct via the builder.

Reply:
I guess I'm willing to believe that those properties happen to align in practice. Let's be careful to watch out for surprising interactions with future extensions.

Reply:
Perhaps I should still comment - or generalize - that the `size` predicate exists for a specific path in the builder that supplies `&[u8]` construction for pointers/arrays of `c_void`. Those get their lifetime extended even if their `basetype` is "an opaque type".

Reply:
Wouldn't hurt! Using `void*` that way is definitely a funny C-ism that might not be obvious.